### PR TITLE
add option `hostname` to create alias API endpoint

### PR DESCRIPTION
**Pull request by thorpelawrence**
Hey there,

Thanks for all the work on this very helpful service.

I've attempted (first time using PHP so might have missed some things, mainly having trouble with the PHP and composer installed in Codespaces) to add a new `hostname` field to the `POST api/v1/aliases` endpoint. The idea being that the main part of the domain name can be prepended to the alias (e.g. `anonaddy.zxcvbn123@uname.anonaddy.me`).

The goal I had in mind is to use this with the Bitwarden extension to automatically include the website name in the alias if desired. [Example of how it's done in their extension](https://github.com/bitwarden/clients/blob/da5e4a57d026e0d09356ad3405e08fb140462fd7/libs/common/src/services/usernameGeneration.service.ts#L187).

PS: are there any steps for contributing that I might have missed, or is it just a case of following the `SELF-HOSTING.md` file mainly?

### CODE REVIEW

File: app/Http/Controllers/Api/AliasController.php [hunk 1]
```diff
@@ -80,32 +86,42 @@ public function store(StoreAliasRequest $request)
                 'extension' => $extension ?? null
             ];
         } else {
+            $prefix = null;
+            if (isset($request->validated()['hostname'])) {
+                // TODO this should be cached, perhaps on boot, from https://publicsuffix.org/
+                // see https://github.com/jeremykendall/php-domain-parser
+                $publicSuffixList = Rules::fromPath(sys_get_temp_dir() . '/public_suffix_list.dat');
```
Comment (thorpelawrence) [inline app/Http/Controllers/Api/AliasController.php:93]: Need to work out a way to do this, wasn't sure if there was anything else that might be cached in a similar way?

(Also doesn't seem like PHP has an equivalent to [path.Join](https://pkg.go.dev/path#Join) so hopefully this is platform agnostic.)

**Comment by willbrowningme** [comment]
If you wish to create aliases of the format `anonaddy.zxcvbn123@uname.anonaddy.me` then you do not need to create these beforehand (assuming you have catch-all enabled for that username).

You can just enter `whatever.zxcvbn123@uname.anonaddy.me` when signing up.

Does Bitwarden also not already include the website name in the alias description?

**Comment by thorpelawrence** [comment]
> If you wish to create aliases of the format `anonaddy.zxcvbn123@uname.anonaddy.me` then you do not need to create these beforehand (assuming you have catch-all enabled for that username).

It can be convenient to have the password manager to generate an alias and password in one flow. I considered making the change with Bitwarden but then it's limited to one provider, and this way there is a single point of formatting for domain names, and being a server rather than frontend code, was hoping it should be easier to handle the domain name processing.

> Does Bitwarden also not already include the website name in the alias description?

It does in the description, however it can be handy to have the website name in the email so you can immediately tell who the sender is in your email inbox, without having to go to the anonaddy site.

**Comment by willbrowningme** [comment]
But then even if this new `hostname` option is added to the create new alias endpoint, Bitwarden still won't post that `hostname` value in their request so the website name will not be included in the alias.

**Comment by thorpelawrence** [comment]
> But then even if this new `hostname` option is added to the create new alias endpoint, Bitwarden still won't post that `hostname` value in their request so the website name will not be included in the alias.

Yeah sorry, I would just add it to Bitwarden as an option, but anything else (the anonaddy extension, app etc) could also send it optionally.

**Pull request closed without merging**